rules/windows/execution_suspicious_powershell_imgload.toml

```
@@ -59,94 +59,45 @@ Attackers can use PowerShell without having to execute `PowerShell.exe` directly
-```
-
-## Setup
-
-If enabling an EQL rule on a non-elastic-agent index (such as beats) for versions <8.2, events will not define `event.ingested` and default fallback for EQL rules was not added until 8.2, so you will need to add a custom pipeline to populate `event.ingested` to @timestamp for this rule to work.
 """
 risk_score = 47
 rule_id = "852c1f19-68e8-43a6-9dce-340771fe1be3"
 severity = "medium"
-tags = ["Elastic", "Host", "Windows", "Threat Detection", "Execution", "Investigation Guide", "Elastic Endgame"]
+tags = ["Elastic", "Host", "Windows", "Threat Detection", "Execution", "Investigation Guide"]
 timestamp_override = "event.ingested"
 type = "eql"
 
 query = '''
-any where host.os.type == "windows" and (event.category : ("library", "driver") or (event.category == "process" and event.action : "Image loaded*")) and
- (dll.name : ("System.Management.Automation.ni.dll", "System.Management.Automation.dll") or
-  file.name : ("System.Management.Automation.ni.dll", "System.Management.Automation.dll")) and
-
-/* add false positives relevant to your environment here */
-not process.executable : ("C:\\Windows\\System32\\RemoteFXvGPUDisablement.exe", "C:\\Windows\\System32\\sdiagnhost.exe") and
-not process.executable regex~ """C:\\Program Files( \(x86\))?\\*\.exe""" and
-  not process.name :
+library where host.os.type == "windows" and
+  dll.name : ("System.Management.Automation.ni.dll", "System.Management.Automation.dll") and
+  not 
   (
-    "Altaro.SubAgent.exe",
-    "AppV_Manage.exe",
-    "azureadconnect.exe",
-    "CcmExec.exe",
-    "configsyncrun.exe",
-    "choco.exe",
-    "ctxappvservice.exe",
-    "DVLS.Console.exe",
-    "edgetransport.exe",
-    "exsetup.exe",
-    "forefrontactivedirectoryconnector.exe",
-    "InstallUtil.exe",
-    "JenkinsOnDesktop.exe",
-    "Microsoft.EnterpriseManagement.ServiceManager.UI.Console.exe",
-    "mmc.exe",
-    "mscorsvw.exe",
-    "msexchangedelivery.exe",
-    "msexchangefrontendtransport.exe",
-    "msexchangehmworker.exe",
-    "msexchangesubmission.exe",
-    "msiexec.exe",
-    "MsiExec.exe",
-    "noderunner.exe",
-    "NServiceBus.Host.exe",
-    "NServiceBus.Host32.exe",
-    "NServiceBus.Hosting.Azure.HostProcess.exe",
-    "OuiGui.WPF.exe",
-    "powershell.exe",
-    "powershell_ise.exe",
-    "pwsh.exe",
-    "SCCMCliCtrWPF.exe",
-    "ScriptEditor.exe",
-    "ScriptRunner.exe",
-    "sdiagnhost.exe",
-    "servermanager.exe",
-    "setup100.exe",
-    "ServiceHub.VSDetouredHost.exe",
-    "SPCAF.Client.exe",
-    "SPCAF.SettingsEditor.exe",
-    "SQLPS.exe",
-    "telemetryservice.exe",
-    "UMWorkerProcess.exe",
-    "w3wp.exe",
-    "wsmprovhost.exe"
+    /* MS Signed Binaries */
+    (
+      process.code_signature.subject_name : (
+        "Microsoft Windows",
+        "Microsoft Dynamic Code Publisher",
+        "Microsoft Corporation"
+      ) and process.code_signature.trusted == true
+    ) or
```

this will miss MS signed lolbins that allow to run malware as a DLL : 

```suggestion
    /* MS Signed Binaries */
    (
      process.code_signature.subject_name : (
        "Microsoft Windows",
        "Microsoft Dynamic Code Publisher",
        "Microsoft Corporation"
      ) and process.code_signature.trusted == true and not process.name : ("rundll32.exe", "regsvr32.exe")
    ) or
```